tests/Schema/ComparatorTest.php

```diff
@@ -662,22 +662,6 @@ public function testCompareForeignKeyBasedOnPropertiesNotName(): void
         self::assertNull($tableDiff);
     }
 
-    public function testCompareForeignKeyRestrictNoActionAreTheSame(): void
-    {
-        $fk1 = new ForeignKeyConstraint(['foo'], 'bar', ['baz'], 'fk1', ['onDelete' => 'NO ACTION']);
-        $fk2 = new ForeignKeyConstraint(['foo'], 'bar', ['baz'], 'fk1', ['onDelete' => 'RESTRICT']);
-
-        self::assertFalse($this->comparator->diffForeignKey($fk1, $fk2));
-    }
-
-    public function testCompareForeignKeyNamesUnqualifiedAsNoSchemaInformationIsAvailable(): void
-    {
-        $fk1 = new ForeignKeyConstraint(['foo'], 'foo.bar', ['baz'], 'fk1');
-        $fk2 = new ForeignKeyConstraint(['foo'], 'baz.bar', ['baz'], 'fk1');
-
-        self::assertFalse($this->comparator->diffForeignKey($fk1, $fk2));
-    }
```

Comment:
This test is supposed to cover the fix (411d38f17264226d53e5b251edff837b8738aab9) for https://github.com/doctrine/dbal/issues/1694 (I can't really find much about the actual issue).

This is obviously a hack. The fact that the schema name of the foreign key is unavailable during schema comparison is a flaw in the API, and doesn't mean that any two foreign with the same name from different schemas are the same key.

While we're keeping the code, I'll remove this test as not worth reworking.